corehq/motech/repeaters/models.py

```diff
@@ -374,25 +375,31 @@ def get_attempt_info(self, repeat_record):
         return None
 
     def register(self, payload):
+        """
+        Registers a SQLRepeatRecord.
+        """
         if not self.allowed_to_forward(payload):
             return
 
-        now = datetime.utcnow()
-        repeat_record = RepeatRecord(
-            repeater_id=self.get_id,
-            repeater_type=self.doc_type,
+        if SQLRepeatRecord.objects.filter(
+            domain=self.domain,
+            repeater_stub=self.repeater_stub,
+            payload_id=payload.get_id,
+            state__in=(RECORD_PENDING_STATE, RECORD_FAILURE_STATE),
+        ).exists():
+            # Payload is already waiting to be sent
+            return
+
+        self.repeater_stub.repeat_records.create(
```

Comment:
I think the model already exists for this specific PR, but if it's within scope, I'll advocate for renaming RepeaterStub, as Stub already has a well-defined meaning within testing.